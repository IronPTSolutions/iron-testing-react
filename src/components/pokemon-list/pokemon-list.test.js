import { render, screen, waitFor } from "@testing-library/react";
import PokemonList from "./pokemon-list";
import pokemonService from "../../services/pokemon-service";

describe("<PokemonList/>", () => {
  test("renders Loading... when pokemons is null", () => {
    // Given
    const mock = jest
      .spyOn(pokemonService, "getPokemons")
      .mockResolvedValue(null);

    // When
    render(<PokemonList />);

    // Then
    expect(mock).toHaveBeenCalled();
    expect(screen.getByText("Loading...")).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });

  test("renders some pokemons", async () => {
    // Given
    const mock = jest
      .spyOn(pokemonService, "getPokemons")
      .mockResolvedValue([{ name: "Bulbasur" }, { name: "Charmander" }]);

    // When

    render(<PokemonList />);

    // Then

    await waitFor(() =>
      expect(screen.queryByText("Loading...")).not.toBeInTheDocument()
    );

    expect(mock).toHaveBeenCalled();
    expect(screen.getByText("Charmander")).toBeInTheDocument();
    expect(screen.getByText("Bulbasur")).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });
});
