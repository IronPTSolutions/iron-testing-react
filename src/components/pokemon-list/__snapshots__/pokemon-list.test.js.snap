// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonList/> renders Loading... when pokemons is null 1`] = `
<body>
  <div>
    <div>
      Loading...
    </div>
  </div>
</body>
`;

exports[`<PokemonList/> renders some pokemons 1`] = `
<body>
  <div>
    <div
      id="pokemon-list"
    >
      <div>
        Bulbasur
      </div>
      <div>
        Charmander
      </div>
    </div>
  </div>
</body>
`;
